torch/nn/quantized/modules/functional_modules.py

```diff
@@ -0,0 +1,76 @@
+import torch
+from torch._ops import ops
+
+
+_VALID_OP_NAMES = {
+    'add',
+    'cat'
+}
+
+class FloatFunctional(torch.nn.Module):
+    def __init__(self, op_name):
+        super(FloatFunctional, self).__init__()
+        if op_name not in _VALID_OP_NAMES:
+            raise NotImplementedError("{} not supported.".format(op_name))
+        self.observer = torch.nn.Identity()
+        self.op_name = op_name
+
+    def forward(self, x):
+        raise RuntimeError("FloatFunctional is not intended to use the " +
+                           "'forward'. Please use the underlying operation")
+
+    def add(self, x, y):
+        # type: (Tensor, Tensor) -> Tensor
+        assert self.op_name == 'add', \
+            "Running add while initialized as {}".format(self.op_name)
+        r = torch.add(x, y)
+        self.observer(r)
+        return r
+
+    def cat(self, x, dim=None):
+        # type: (List[Tensor], Optional[int]) -> Tensor
+        assert self.op_name == 'cat', \
+            "Running cat while initialized as {}".format(self.op_name)
+        if dim is None:
```

Comment:
add a TODO to fix op registration. This logic here feels out of place

Reply:
The registtration will support optional arguments. The thing is that I still believe we should resolve it here, instead of doing it  inside of the kernel. The reason is that Once the documentation is parsed, the default values will be shown. But if we do it in the kernel, the default value will be hidden from the user.

Reply:
I can make it optional with default value of 0

Reply:
If it's a bug - then file an issue against JIT/op bindings to fix it :) It feels out of place here.

Reply:
I don't think it's a bug, as it is documented in the [Type Refinement](https://pytorch.org/docs/stable/jit.html#optional-type-refinement) section of the jit docs. @jamesr66a Is there a way to use optional args, while dropping the `if None` test?

Reply:
Actually, you're right, it's not a optional arg, it's an arg with default value: https://github.com/pytorch/pytorch/blob/77c08aa46c3f3460b95b89cbe357b99180bc824d/aten/src/ATen/native/native_functions.yaml#L399

Then why you just don't duplicate the signature to be the same, i.e. `int dim = 0`?

Reply:
I think the ideal solution in python would be to use kwargs and forward them directly:

```
import torch

dim = None

x = [torch.rand(3, 4) for i in range(3)]

def cat(x, **kwargs):
    torch.cat(x, **kwargs)

cat(x, dim=1)
cat(x)
```

However, this isn't supported by script, so what's used here is fine. @zafartahirov also proposed making the default `dim` equal to ` in the function signature here, which would probably also work